tests/x11/multi_users_dm.pm

```diff
@@ -77,6 +77,7 @@ sub run {
     handle_login($user, 1);
     handle_welcome_screen(timeout => 120) if (opensuse_welcome_applicable);
     assert_screen 'generic-desktop', 60;
```

Comment:
IMO this should rather do:

```
assert_screen qw(generic-desktop gnome-activities), 60;
```

as other places which assert `generic-desktop` don't expect the activities view either.

Reply:
the gnome-activites tag in generic desktop is already used in quite a few places, most notably in wait_boot_past_bootloader

Reply:
According to my grep, in exactly those two only :P

```
lib/opensusebasetest.pm:    send_key('esc') if match_has_tag('gnome-activities');
tests/x11/user_gui_login.pm:    send_key('esc') if match_has_tag('gnome-activities');    # GNOME 40 logs in with activities opened
```

Reply:
the one in lib/opensusebasetest also uses gnome-activites as extra-tag on opensuse-welcome

as on login, there can be an app pre-started, which would start in activities view too

Reply:
FWICT, `gnome-activities` should only be visible directly after login. So `multi_users_dm` and `wait_boot_past_bootloader` are the most prominent cases for that, are there others? I also see `tests/x11/remote_desktop/onetime_vncsession_xvnc_tigervnc.pm` calling `handle_login`  followed by `assert_screen 'generic-desktop'`

Reply:
That should be the most prominent parts I'd also see. the most common run is first_boot, which calls out to wait_boot_past_bootloader

Reply:
I can see another case for that:  `handle_relogin` invokes `handle_logout `and `handle_login`, and `tests/x11/gnomecase/application_starts_on_login.pm` calling `handle_relogin` followed by `assert_screen "generic-desktop";`
Since I was working on fixing application_starts_on_login failures, I will fix it later.

Reply:
> I can see another case for that: `handle_relogin` invokes `handle_logout `and `handle_login`, and `tests/x11/gnomecase/application_starts_on_login.pm` calling `handle_relogin` followed by `assert_screen "generic-desktop";`
> Since I was working on fixing application_starts_on_login failures, I will fix it later.

handle_login should take care of reaching the desktop (i.e. do the same as wait_boot_past_bootloader)
This sounds we're heading for a bit of restructuring the code heare

Reply:
Yeah, you are right. 
So, will you restructure the code to let handle_login take care of reaching the desktop?

Reply:
@GraceWang571 
 Not sure if I get to that anytime soon - I am injecting some openQA work as time permits. If 
you have some spare cycles, I'd highly appreciate if you could take that one

Reply:
> @GraceWang571
> Not sure if I get to that anytime soon - I am injecting some openQA work as time permits. If
> you have some spare cycles, I'd highly appreciate if you could take that one

@DimStar77  Sorry for the late reply. 
I'd be very glad to take this one (I mean take care of reaching the desktop in handle_login)

Reply:
Is there a PR pending meanwhile? I couldn't find any, so I made one with just a quick fix for the `login_test` failure: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12858

Reply:
Thanks for the quick fix for login_test.
I have a two-day training today and tomorrow, so no new PR for the restructure now.